providers/mailjet/src/lib/mailjet.provider.ts

```diff
@@ -72,4 +74,59 @@ export class MailjetEmailProvider implements IEmailProvider {
       code: CheckIntegrationResponseEnum.SUCCESS,
     };
   }
+  
+  getMessageId(body: any | any[]): string[] {
+    if (Array.isArray(body)) {
+      return body.map((item) => item.MessageID);
+    }
+
+    return [body.MessageID];
+  }
+  
+  parseEventBody(
+    body: any | any[],
+    identifier: string
+  ): IEmailEventBody | undefined {
+    if (Array.isArray(body)) {
+      body = body.find((item) => item.MessageID === identifier);
+    }
+
+    if (!body) {
+      return undefined;
+    }
+
+    const status = this.getStatus(body.event);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body.MessageID,
+      attempts: body.attempt ? parseInt(body.attempt, 10) : 1,
+      response: body.response ? body.response : '',
+      row: body,
+    };
+  }
+  
+  private getStatus(event: string): EmailEventStatusEnum | undefined {
+    switch (event) {
+      case 'open':
+        return EmailEventStatusEnum.OPENED;
+      case 'bounce':
+        return EmailEventStatusEnum.BOUNCED;
+      case 'click':
+        return EmailEventStatusEnum.CLICKED;
+      case 'sent':
+        return EmailEventStatusEnum.SENT;
+      case 'blocked':
+        return EmailEventStatusEnum.BLOCKED;
+      case 'spam':
+        return EmailEventStatusEnum.SPAM;
+      case 'unsub':
+        return EmailEventStatusEnum.UNSUBSCRIBE;
+    }
```

Comment:
I'd suggest to do an enum with the statused of Mailjet rather than hardcode them. Would help to understand where these strings come from.

Reply:
A trivial question. Mailjet shares some statuses in common with the provider statuses, i.e, Open, Bounce, Click. What about them?

Reply:
What I meant is to do an enum like:
```
enum MailjetStatusEnum {
  OPEN = 'open',
  SPAM = 'spam',
  UNSUB = 'unsub'
  ...
}
```
to replace the hardcode checks of the switch case for something more manageable in case we have to use it elsewhere and anything changes on Mailjet side.

Reply:
So you mean we should create dedicated enum for different providers, right? So which file should we place it in?

Reply:
In `mailjet.provider.ts` would be fine. The same as I mentioned in the Mandrill PR. 🙂